src/cool_seq_tool/mappers/exon_genomic_coords.py

```diff
@@ -754,188 +747,163 @@ async def _genomic_to_tx_segment(
-                    if err_msg:
-                        return GenomicTxSeg(errors=[err_msg])
-                    gene = _gene
+        # Select a transcript if not provided
+        if not transcript:
+            mane_transcripts = self.mane_transcript_mappings.get_gene_mane_data(gene)
 
-                return GenomicTxSeg(
-                    gene=gene,
-                    genomic_ac=genomic_ac,
-                    tx_ac=transcript,
-                    seg=TxSegment(
-                        exon_ord=exon_num,
-                        offset=offset,
-                        genomic_location=genomic_location,
-                    ),
+            if mane_transcripts:
+                transcript = mane_transcripts[0]["RefSeq_nuc"]
+            else:
+                # Attempt to find a coding transcript if a MANE transcript
+                # cannot be found
+                results = await self.uta_db.get_transcripts(
+                    gene=gene, alt_ac=genomic_ac
                 )
 
-        if genomic_ac:
-            _gene, err_msg = await self._get_genomic_ac_gene(genomic_pos, genomic_ac)
+                if not results.is_empty():
+                    transcript = results[0]["tx_ac"][0]
+                else:
+                    # Run if gene is for a noncoding transcript
+                    query = f"""
+                        SELECT DISTINCT tx_ac
+                        FROM {self.uta_db.schema}.tx_exon_aln_v
+                        WHERE hgnc = '{gene}'
+                        AND alt_ac = '{genomic_ac}'
+                        """  # noqa: S608
+                    result = await self.uta_db.execute_query(query)
+
+                    if result:
+                        transcript = result[0]["tx_ac"]
+                    else:
+                        return GenomicTxSeg(
+                            errors=[
+                                f"Could not find a transcript for {gene} on {genomic_ac}"
+                            ]
+                        )
+        tx_exons = await self._get_all_exon_coords(
+            tx_ac=transcript, genomic_ac=genomic_ac
+        )
+        if not tx_exons:
+            return GenomicTxSeg(errors=[f"No exons found given {transcript}"])
+
+        strand = Strand(tx_exons[0].alt_strand)
+        params["strand"] = strand
+        use_alt_start_i = self._use_alt_start_i(
+            is_seg_start=is_seg_start, strand=strand
+        )
+        if use_alt_start_i and coordinate_type == CoordinateType.RESIDUE:
+            genomic_pos = genomic_pos - 1  # Convert residue coordinate to inter-residue
 
+        # gene is not required to liftover coordinates if tx_ac and genomic_ac are given, but we should set the associated gene
+        if not gene:
+            _gene, err_msg = await self._get_tx_ac_gene(transcript)
             if err_msg:
                 return GenomicTxSeg(errors=[err_msg])
+            gene = _gene
```

Comment:
It's probably best to move this section immediately after we get a `transcript`. That way we can fail fast